src/hooks/useActionCenterSubscriber/useLimitOrderActionSubscriber.tsx

```diff
@@ -0,0 +1,394 @@
+  selectActiveQuoteId,
+  selectActiveQuoteLimitPrice,
+  selectActiveQuoteSellAsset,
+  selectLimitOrderSubmissionMetadata,
+} from '@/state/slices/limitOrderSlice/selectors'
+import {
+  selectLimitOrderActionByCowSwapQuoteId,
+  selectLimitOrderActionByLimitOrderId,
+  selectOpenLimitOrderActionsFilteredByWallet,
+} from '@/state/slices/selectors'
+import { store, useAppDispatch, useAppSelector, useSelectorWithArgs } from '@/state/store'
+
+type UseLimitOrderActionSubscriberProps = {
+  onDrawerOpen: () => void
+}
+
+export const useLimitOrderActionSubscriber = ({
+  onDrawerOpen,
+}: UseLimitOrderActionSubscriberProps) => {
+  const dispatch = useAppDispatch()
+  const translate = useTranslate()
+  const toastColor = useColorModeValue('white', 'gray.900')
+
+  const {
+    number: { toCrypto },
+  } = useLocaleFormatter()
+
+  const activeQuote = useAppSelector(limitOrderSlice.selectors.selectActiveQuote)
+  const sellAsset = useAppSelector(selectActiveQuoteSellAsset)
+  const buyAsset = useAppSelector(selectActiveQuoteBuyAsset)
+  const sellAmountCryptoBaseUnit = useAppSelector(selectInputSellAmountCryptoBaseUnit)
+  const buyAmountCryptoBaseUnit = useAppSelector(selectBuyAmountCryptoBaseUnit)
+  const quoteId = useAppSelector(selectActiveQuoteId)
+  const { currentData: ordersResponse } = useLimitOrdersQuery()
+  const toast = useToast({
+    render: ({ title, status, description, onClose }) => {
+      const handleClick = () => {
+        onClose()
+        onDrawerOpen()
+      }
+
+      const toastSx = {
+        '&:hover': {
+          cursor: 'pointer',
+          background: status === 'success' ? 'green.300' : 'red.300',
+        },
+      }
+
+      return (
+        <Flex
+          // We can't memo this because onClose prop comes from the render props
+          // eslint-disable-next-line react-memo/require-usememo
+          onClick={handleClick}
+          background={status === 'success' ? 'green.500' : 'red.500'}
+          color='white'
+          px={4}
+          py={2}
+          borderRadius='md'
+          // eslint-disable-next-line react-memo/require-usememo
+          sx={toastSx}
+        >
+          <Box py={1} me={2}>
+            {status === 'success' ? (
+              <Icon color={toastColor} as={TbCircleCheckFilled} height='20px' width='20px' />
+            ) : (
+              <Icon color={toastColor} as={TbCircleXFilled} height='20px' width='20px' />
+            )}
+          </Box>
+          <Box>
+            <CText fontWeight='bold' color={toastColor}>
+              {title}
+            </CText>
+            <CText color={toastColor}>{description}</CText>
+          </Box>
+        </Flex>
+      )
+    },
+  })
+  const openLimitOrders = useAppSelector(selectOpenLimitOrderActionsFilteredByWallet)
+  const quoteExpirationTimestamp = useAppSelector(selectActiveQuoteExpirationTimestamp)
+
+  const orderSubmissionMetadataFilter = useMemo(() => {
+    return { cowSwapQuoteId: quoteId ?? 0 }
```

Comment:
q: dis safe?

Reply:
Done in [2712581](https://github.com/shapeshift/web/pull/9626/commits/271258198517d54466a30d742e57de2d7e53f15b)

Even if the whole limit order feature is using the same "scary pattern"